spring-ai-core/src/main/java/org/springframework/ai/chat/client/advisor/api/AfterAdvisor.java

```diff
@@ -77,21 +72,18 @@ public enum StreamResponseMode {
 
 	}
 
-	default StreamResponseMode getStreamResponseMode() {
-		return StreamResponseMode.ON_FINISH_ELEMENT;
+	default AfterStreamMode getAfterStreamMode() {
+		return AfterStreamMode.ON_FINISH_ELEMENT;
 	}
 
 	/**
-	 * @param fluxResponse the streaming {@link ChatResponse} data to be advised.
+	 * @param advisedResponseStream the streaming {@link ChatResponse} data to be advised.
 	 * Represents the row {@link ChatResponse} stream data after the
 	 * {@link ChatModel#stream(Prompt)} method is called.
-	 * @param adviseContext the shared data between the advisors in the chain. It is
-	 * shared between all request and response advising points of all advisors in the
-	 * chain.
 	 * @return the advised {@link ChatResponse} flux.
 	 */
-	default Flux<ChatResponse> adviseResponse(Flux<ChatResponse> fluxResponse, Map<String, Object> adviseContext) {
-		return fluxResponse;
+	default Flux<AdvisedResponse> afterStream(Flux<AdvisedResponse> advisedResponseStream) {
```

Comment:
This API feels that it should belong elsewhere, to a specific custom advisor type. Currently, from my understanding, this is only called for the custom case. For most regular advisors this would be confusing that this can be overridden.